src/api/indicators/IndicatorAPI.js

```diff
@@ -62,11 +77,20 @@ class IndicatorAPI extends EventEmitter {
    * myIndicator.text("Hello World!");
    * myIndicator.iconClass("icon-info");
    *
+   * If you would like to use a Vue component, you can pass it in
+   * directly as the 'vueComponent' attribute of the indicator object.
+   * This accepts a Vue component or a promise that resolves to a Vue component (for asynchronous
+   * rendering).
+   *
+   * @param {Indicator} indicator
    */
   add(indicator) {
     if (!indicator.priority) {
       indicator.priority = this.openmct.priority.DEFAULT;
     }
+    if (!indicator.vueComponent) {
+      indicator.vueComponent = vueWrapHtmlElement(indicator.element);
```

Comment:
Is the `element` property for indicator objects exist after removing the `mount` utility? See here: https://github.com/nasa/openmct/blob/vue-indicators/src/plugins/clearData/plugin.js#L34

Perhaps you need to create the `indicator.element` node here as well?

Reply:
@shefalijoshi Not for most of our default indicators. The idea is that if the `vueComponent` is provided to the indicator API, it will mount the component directly. Otherwise, it will look for the `element` property and wrap it in an anonymous Vue component. So one or the other is required here. If both vueComponent and element are missing, then we fail fast.

Reply:
OK. Add some documentation around this in the API change.

Reply:
are the [existing docs ](https://github.com/nasa/openmct/pull/7395/files/a5db40d4e1a37c3da524780ddd6283a4fb88697b#diff-426ef19cfbf417b2b9ba4e46dc02a171d3549d2f32550f3dd346461c81633f1cR80)ok? this just explains how to use the API, the implementation details shouldn't be important to the average user?